libevmasm/SimplificationRules.cpp

```diff
@@ -101,9 +101,14 @@ Rules::Rules()
 		{{Instruction::SDIV, {A, B}}, [=]{ return B.d() == 0 ? 0 : s2u(divWorkaround(u2s(A.d()), u2s(B.d()))); }},
 		{{Instruction::MOD, {A, B}}, [=]{ return B.d() == 0 ? 0 : modWorkaround(A.d(), B.d()); }},
 		{{Instruction::SMOD, {A, B}}, [=]{ return B.d() == 0 ? 0 : s2u(modWorkaround(u2s(A.d()), u2s(B.d()))); }},
-		{{Instruction::EXP, {A, B}}, [=]{ return u256(boost::multiprecision::powm(bigint(A.d()), bigint(B.d()), bigint(1) << 256)); }},
+		{{Instruction::EXP, {A, B}}, [=]{
+			// safety case against compile time bomb
+			if (A.d() >= 2 && B.d() >= 256)
+				return ~u256(0);
```

Comment:
Hm, right, what does the YP say in case the value cannot be represented in 256-bit? Will the calculation need to be done properly and the result mod 256 is stored on the stack?

This is theoretical though because it is limited by gas.